presto-hive-common/src/main/java/org/apache/hadoop/fs/HadoopExtendedFileSystem.java

```diff
@@ -65,6 +65,12 @@ public FileSystem getRawFileSystem()
         return fs;
     }
 
+    @Override
+    public String getScheme()
+    {
+        return fs.getScheme();
```

Comment:
Why do we need this API?

Reply:
If we dont fix this method, MOR tables could not be queried.  I added the intergration tests in https://github.com/prestodb/presto/pull/17463.
Following is part of the error messages:
> Caused by: java.lang.UnsupportedOperationException: Not implemented by the HadoopExtendedFileSystem FileSystem implementation
	at org.apache.hadoop.fs.FileSystem.getScheme(FileSystem.java:219)
	at org.apache.hudi.common.fs.FSUtils.isGCSFileSystem(FSUtils.java:592)
	at org.apache.hudi.common.table.log.HoodieLogFileReader.getFSDataInputStream(HoodieLogFileReader.java:119)
	at org.apache.hudi.common.table.log.HoodieLogFileReader.<init>(HoodieLogFileReader.java:95)
	at org.apache.hudi.common.table.log.HoodieLogFileReader.<init>(HoodieLogFileReader.java:86)
	at org.apache.hudi.common.table.log.HoodieLogFormat.newReader(HoodieLogFormat.java:282)
	at org.apache.hudi.common.table.log.LogReaderUtils.readSchemaFromLogFileInReverse(LogReaderUtils.java:49)
	at org.apache.hudi.common.table.log.LogReaderUtils.readLatestSchemaFromLogFiles(LogReaderUtils.java:77)
	at org.apache.hudi.hadoop.realtime.AbstractRealtimeRecordReader.init(AbstractRealtimeRecordReader.java:85)
	at org.apache.hudi.hadoop.realtime.AbstractRealtimeRecordReader.<init>(AbstractRealtimeRecordReader.java:67)
	at org.apache.hudi.hadoop.realtime.RealtimeCompactedRecordReader.<init>(RealtimeCompactedRecordReader.java:62)
	at org.apache.hudi.hadoop.realtime.HoodieRealtimeRecordReader.constructRecordReader(HoodieRealtimeRecordReader.java:70)
	at org.apache.hudi.hadoop.realtime.HoodieRealtimeRecordReader.<init>(HoodieRealtimeRecordReader.java:47)
	at org.apache.hudi.hadoop.realtime.HoodieParquetRealtimeInputFormat.getRecordReader(HoodieParquetRealtimeInputFormat.java:323)
	at com.facebook.presto.hive.HiveUtil.createRecordReader(HiveUtil.java:272)
	at com.facebook.presto.hive.GenericHiveRecordCursorProvider.lambda$createRecordCursor$0(GenericHiveRecordCursorProvider.java:74)
	at com.facebook.presto.hive.authentication.NoHdfsAuthentication.doAs(NoHdfsAuthentication.java:23)
	at com.facebook.presto.hive.HdfsEnvironment.doAs(HdfsEnvironment.java:81)
	at com.facebook.presto.hive.GenericHiveRecordCursorProvider.createRecordCursor(GenericHiveRecordCursorProvider.java:73)
	at com.facebook.presto.hive.HivePageSourceProvider.getPageSourceFromCursorProvider(HivePageSourceProvider.java:571)
	 
You can check the full error log at https://github.com/prestodb/presto/runs/5566744421?check_suite_focus=true.

Reply:
BTW, please review the PR at https://github.com/prestodb/presto/pull/17463

Reply:
Got it.